api/middleware/cors.go

```diff
@@ -5,14 +5,18 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
+var (
+	allowedOrigins = []string{"https://temporal.cloud"}
+)
+
 // CORSMiddleware is used to load our CORS handling logic
-// TODO: need to cleanup and restrict origins, the all origin allow
-// and credentials disable is a temporary measure
 func CORSMiddleware() gin.HandlerFunc {
 	corsConfig := cors.DefaultConfig()
-	corsConfig.AllowAllOrigins = true
-	corsConfig.AllowCredentials = false
+	// configure allowed origins
+	corsConfig.AllowOrigins = allowedOrigins
```

Comment:
could allow all for ease of dev work with dev mode enabled

Reply:
good call